torch_xla/experimental/pjrt.py

```diff
@@ -179,9 +192,25 @@ def _thread_fn(device: torch.device):
 
     return fn()
 
+  # TODO(wcromar): remove this when the TPU master IP becomes predictable
+  def _discover_tpu_master_worker_ip(device: torch.device):
+    torch_xla._XLAC._xla_set_default_device(device)
+
+    return tpu.discover_master_worker_ip()
+
   with concurrent.futures.ThreadPoolExecutor(
       max_workers=num_threads) as executor:
-    # TODO: clean up this statement
+
+    if os.getenv('PJRT_INIT_TORCH_DISTRIBUTED', '0') == '1':
+      if device_type() == 'TPU':
+        # HACK: need to call with each device since it relies on an XLA collective
+        master_ip = next(executor.map(_discover_tpu_master_worker_ip, devices))
+        init_method = f'tcp://{master_ip}:{master_port}'
+      else:
+        init_method = None
+
+      init_pjrt_process_group(init_method=init_method)
```

Comment:
hmm, so user don't need to init the process group if they are using our wrapper?

Reply:
See above.